### ISSUE TITLE
Absolute file path as partition key overwrites original file

### ISSUE THREAD

**Issue by jcbiddle** [dagster-io/dagster#16064]
### Dagster version

dagster, version 1.4.3

### What's the issue?

I'm exploring the use of dynamic partitions and found that if the absolute path to an input file is used as the partition key, then the original input file will be overwritten by the default IO manager. The issue stems from the way that the output location is computed in the `UPathIOManager._get_paths_for_partitions` method. The problem is that the final path is computed as 

```path = <asset_path> / <partition_key>```

According to the [Pathlib spec](https://docs.python.org/3/library/pathlib.html#operators), this will just return the partition key if it is an absolute path.

This may be intended behaviour, but is probably at least worth a mention in in the docs to ensure users don't unwittingly overwrite their data. Alternatively, it may be desirable to separate the partition key and input configuration to enable users to more easily supply per-partition configuration.

### Deployment type

None

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by anotine10** [comment]
I do not think it is an expected behaviour, and it may me think that Partition has not been though to be used to encode  a list of file paths. Even with Relative nodes, it finally saved the output in a folder not really expected. 
Is there any example in dagster that use partition to separate an asset in a list of files? 
Thank you and kudos for the great work!

**Comment by maximearmstrong** [comment]
Hi @anotine10 - Indeed, that's not the expected behavior - we think the bug is introduced with the implementation using the `/` operator, like mentioned before by @jcbiddle. Updating the implementation should resolve this issue.
